src/net.h

```diff
@@ -613,7 +613,7 @@ class CNode
      * closed.
      * Otherwise this unique_ptr is empty.
      */
-    std::unique_ptr<i2p::sam::Session> m_i2p_sam_session GUARDED_BY(m_sock_mutex);
+    std::unique_ptr<i2p::sam::Session> m_i2p_sam_session GUARDED_BY(m_sock_mutex) PT_GUARDED_BY(m_sock_mutex);
```

Comment:
Question, are both guards useful here? (All but one of the other instances of `PT_GUARDED_BY` don't cumulate both annotations.)  

```diff
-    std::unique_ptr<i2p::sam::Session> m_i2p_sam_session GUARDED_BY(m_sock_mutex) PT_GUARDED_BY(m_sock_mutex);
+    std::unique_ptr<i2p::sam::Session> m_i2p_sam_session PT_GUARDED_BY(m_sock_mutex);
 };
```

Reply:
No, the original annotation was correct; for some reason I thought PT_GUARDED_BY added some checks GUARDED_BY doesn't have, but that doesn't seem to be the case.

Changing from GUARDED_BY to PT_GUARDED_BY alone would allow you to do `m_i2p_sam_session = nullptr` without holding the lock, or `if (m_i2p_sam_session != nullptr) { LOCK(m_sock_mutex); m_i2p_sam_session->foo(); }` which would be racy.